src/components/magnet/field-line.tsx

```diff
@@ -41,10 +41,15 @@ export default PixiComponent<IProps, PIXI.Graphics>("FieldLine", {
       const delta = getFieldVectorAtPosition(magnets, magnetModels, currPos.x, currPos.y).unit();
       currPos = currPos.add(delta);
       instance.lineTo(currPos.x, currPos.y);
-
       if (outOfBounds(currPos)
-          || magnets.some(magnet => pointInMagnet(magnet, currPos.x, currPos.y)) && !internal
-          || !magnets.some(magnet => pointInMagnet(magnet, currPos.x, currPos.y)) && internal
+          || magnets.length && magnetModels.length
+          && (pointInMagnet(magnets[0], magnetModels[0], currPos.x, currPos.y) && !internal)
+          || magnets.length && magnetModels.length
+          && (!pointInMagnet(magnets[0], magnetModels[0], currPos.x, currPos.y) && internal)
+          || magnets.length > 1 && magnetModels.length > 1
+          && (pointInMagnet(magnets[1], magnetModels[1], currPos.x, currPos.y) && !internal)
+          || magnets.length > 1 && magnetModels.length > 1
+          && (!pointInMagnet(magnets[1], magnetModels[1], currPos.x, currPos.y) && internal)
       ) {
```

Comment:
@ekosmin This got really ugly really quickly.  It's a side effect of having some information in the magnet state and some in the simulation state.  Hopefully we can rip this out and clean it up when we move everything into the simulation.

Reply:
This gets ripped out in the field-calculation-2 branch, so don't worry.

Reply:
👍